extension/ackextension/inmemory.go

```diff
@@ -0,0 +1,111 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ackextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/ackextension"
+
+import (
+	"context"
+	"sync/atomic"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+	"go.opentelemetry.io/collector/component"
+)
+
+// InMemoryAckExtension is the in-memory implementation of the AckExtension
+type InMemoryAckExtension struct {
+	partitionMap                  *lru.Cache[string, *ackPartition]
+	maxNumPendingAcksPerPartition uint64
+}
+
+func newInMemoryAckExtension(conf *Config) *InMemoryAckExtension {
+	cache, _ := lru.New[string, *ackPartition](int(conf.MaxNumPartition))
+	return &InMemoryAckExtension{
+		partitionMap:                  cache,
+		maxNumPendingAcksPerPartition: defaultMaxNumPendingAcksPerPartition,
```

Comment:
did you mean `conf.MaxNumPendingAcksPerPartition` instead of `defaultMaxNumPendingAcksPerPartition`

Reply:
ah you're right [face_palm]